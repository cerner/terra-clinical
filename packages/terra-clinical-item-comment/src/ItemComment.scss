@import './mixins';

.terraClinical-ItemComment {
  @include terra-clinical-text-body;
  color: $terra-grey-70;
}

.terraClinical-ItemComment--icon {
  fill: $terra-grey-30;
}
