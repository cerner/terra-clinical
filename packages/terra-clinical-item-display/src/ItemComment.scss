@import './mixins';

:local {
  .item-comment {
    @include terra-clinical-item-body;
    color: #64696c;
  }
}
