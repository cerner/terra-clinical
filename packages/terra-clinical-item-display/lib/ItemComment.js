'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('terra-base/lib/baseStyles');

var _IconComment = require('terra-icon/lib/icon/IconComment');

var _IconComment2 = _interopRequireDefault(_IconComment);

var _ItemDisplay = require('./ItemDisplay');

var _ItemDisplay2 = _interopRequireDefault(_ItemDisplay);

require('./ItemComment.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  /**
   * The text to be displayed for the comment.
   */
  text: _react.PropTypes.string,
  /**
   * Whether or not the text should be truncated in display.
   */
  isTruncated: _react.PropTypes.bool
};

var defaultProps = {
  text: '',
  isTruncated: false
};

var ItemComment = function ItemComment(_ref) {
  var text = _ref.text,
      isTruncated = _ref.isTruncated,
      customProps = _objectWithoutProperties(_ref, ['text', 'isTruncated']);

  var commentClassNames = (0, _classnames2.default)(['terraClinical-ItemComment', customProps.className]);

  var commentIcon = _react2.default.createElement(_IconComment2.default, null);

  return _react2.default.createElement(_ItemDisplay2.default, _extends({
    text: text,
    isTruncated: isTruncated,
    icon: commentIcon
  }, customProps, {
    className: commentClassNames
  }));
};

ItemComment.propTypes = propTypes;
ItemComment.defaultProps = defaultProps;

exports.default = ItemComment;