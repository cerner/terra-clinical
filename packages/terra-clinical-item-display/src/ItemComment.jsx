import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import CommentIndicator from 'terra-icon/lib/icon/IconComment';
import ItemDisplay from './ItemDisplay';
import styles from './ItemComment.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The text to be displayed for the comment.
   */
  text: PropTypes.string,
  /**
   * Whether or not the text should be truncated in display.
   */
  isTruncated: PropTypes.bool,
};

const defaultProps = {
  text: '',
  isTruncated: false,
};

const ItemComment = ({
    text,
    isTruncated,
    ...customProps
  }) => (
    <ItemDisplay
      text={text}
      isTruncated={isTruncated}
      icon={<CommentIndicator />}
      {...customProps}
      className={cx('item-comment', customProps.className)}
    />
  );

ItemComment.propTypes = propTypes;
ItemComment.defaultProps = defaultProps;

export default ItemComment;
