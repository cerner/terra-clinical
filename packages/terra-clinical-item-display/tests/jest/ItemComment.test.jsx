import React from 'react';
import ItemComment from '../../src/ItemComment';

// Snapshot Tests
it('should render a default component', () => {
  const comment = shallow(<ItemComment />);
  expect(comment).toMatchSnapshot();
});

it('should render a comment with text', () => {
  const comment = shallow(<ItemComment text="comment" />);
  expect(comment).toMatchSnapshot();
});

it('should render a truncated comment', () => {
  const comment = shallow(<ItemComment text="comment1comment1comment1comment1comment1comment1comment1comment1" isTruncated />);
  expect(comment).toMatchSnapshot();
});

it('should render a comment with the attention textStyle on text', () => {
  const comment = shallow(<ItemComment text="ItemComment" textStyle="attention" />);
  expect(comment).toMatchSnapshot();
});

it('should render a comment with the secondary textStyle on text', () => {
  const comment = shallow(<ItemComment text="ItemComment" textStyle="secondary" />);
  expect(comment).toMatchSnapshot();
});

it('should render a comment with the strikeThrough textStyle on text', () => {
  const comment = shallow(<ItemComment text="ItemComment" textStyle="strikeThrough" />);
  expect(comment).toMatchSnapshot();
});
