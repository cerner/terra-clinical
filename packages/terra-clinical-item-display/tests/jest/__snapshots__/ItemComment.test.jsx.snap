// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a comment with text 1`] = `
<ItemDisplay
  className="item-comment"
  icon={
    <IconComment
      className=""
      data-name="Layer 1"
      isBidi={true}
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  isTruncated={false}
  text="comment"
/>
`;

exports[`should render a comment with the attention textStyle on text 1`] = `
<ItemDisplay
  className="item-comment"
  icon={
    <IconComment
      className=""
      data-name="Layer 1"
      isBidi={true}
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  isTruncated={false}
  text="ItemComment"
  textStyle="attention"
/>
`;

exports[`should render a comment with the secondary textStyle on text 1`] = `
<ItemDisplay
  className="item-comment"
  icon={
    <IconComment
      className=""
      data-name="Layer 1"
      isBidi={true}
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  isTruncated={false}
  text="ItemComment"
  textStyle="secondary"
/>
`;

exports[`should render a comment with the strikeThrough textStyle on text 1`] = `
<ItemDisplay
  className="item-comment"
  icon={
    <IconComment
      className=""
      data-name="Layer 1"
      isBidi={true}
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  isTruncated={false}
  text="ItemComment"
  textStyle="strikeThrough"
/>
`;

exports[`should render a default component 1`] = `
<ItemDisplay
  className="item-comment"
  icon={
    <IconComment
      className=""
      data-name="Layer 1"
      isBidi={true}
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  isTruncated={false}
  text=""
/>
`;

exports[`should render a truncated comment 1`] = `
<ItemDisplay
  className="item-comment"
  icon={
    <IconComment
      className=""
      data-name="Layer 1"
      isBidi={true}
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  isTruncated={true}
  text="comment1comment1comment1comment1comment1comment1comment1comment1"
/>
`;
